llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6545,6 +6544,19 @@ static bool SwitchToLookupTable(SwitchInst *SI, IRBuilder<> &Builder,
 
   BranchInst *RangeCheckBranch = nullptr;
 
+  // Grow the table to cover all possible index values to avoid the range check.
+  if (UseSwitchConditionAsTableIndex) {
+    ConstantRange CR = computeConstantRange(TableIndex, /* ForSigned */ false);
+    if (SwitchLookupTable::WouldFitInRegister(
```

Comment:
Maybe add a comment explaining why we check `WouldFitInRegister`: that by doing so, growing the table shouldn't have any size impact.

And maybe put a TODO that we could consider growing the table also when it doesn't fit in a register. A small size increase might be worth removing the range check.

Reply:
Apply your comment, thanks